### DISCUSSION TITLE
Predicting from 1 instance, not 1 batch

### DISCUSSION THREAD

**Discussion by dorienh** [Lightning-AI/pytorch-lightning#13729] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
Hi, 

I am deploying a model whereby speed is of the essence. It's a time series model, so it is trained in larger batches, each with x time_steps as input to the actual model (LSTM input size). 

I see the [documentation](https://pytorch-lightning.readthedocs.io/en/latest/deploy/production_basic.html) about standard predicting, e.g. in the simplest case: 

```
model = LitModel.load_from_checkpoint("best_model.ckpt")
model.eval()
x = torch.randn(1, 64) #but then with 1 dimension more for LSTM

with torch.no_grad():
    y_hat = model(x)
```

But is there any way I can avoid to feed the entire batch? I just need to predict for the last time_steps. I fear that it would slow down the model if I need to make many unnecessary predictions. 

Sorry if this is a bit of a newbie question.

**Comment by Klassikcat** [comment]
Hi, I think you just need to convert raw datas into tensor, unsqueeze the tensor(so that it can avoid dimension issue), then pass into model itself. for example(just a pseudocode),

```python
raw_data = # input from any source
tensor = convert_func(raw_data) # your custom codes should be here. tensor size should be like: (n)
tensor = tensor.unsqueeze() # tensor size should become (1, n). normally, batch_size should be in first dimension. since batch_size is 1, it contains only 1 data in tensor.
result = model(tensor)
return result
```

**Reply by dorienh** [reply]
Wouldn't the model expect multiple batches though and output an error? 

The model is trained with batch_size = n, but here wondering if I can predict with only 1 instance instead of a whole batch

**Reply by Klassikcat** [reply]
@dorienh it would not raise dimension error because `torch.Tensor.unsqueeze(dim=0)` adds additional dimension and it will be regarded as batch size if there is no batch size dimension in your tensor instance already.

```python
x = torch.rand((12, 12))
x.shape
```
`out: torch.size(12, 12)`

```python
x_new = x.unsqueeze(dim=0) # it adds dimension at dim 0.
x_new.shape
```
`out: torch.size(1, 12, 12)`
